glotaran/test/test_spectral_decay_full_model.py

```diff
@@ -0,0 +1,223 @@
+import numpy as np
+import pytest
+
+from glotaran.analysis.optimize import optimize
+from glotaran.analysis.simulation import simulate
+from glotaran.io import load_model
+from glotaran.io import load_parameters
+from glotaran.project import Scheme
+
+MODEL_3C_BASE = """\
+dataset:
+    dataset1: &dataset1
+        megacomplex: [mc1]
+        global_megacomplex: [mc2]
+        initial_concentration: j1
+        irf: irf1
+    dataset2: &dataset2
+        megacomplex: [mc2]
+        global_megacomplex: [mc1]
+        initial_concentration: j1
+        irf: irf1
+    dataset3: &dataset3
+        megacomplex: [mc1]
+        initial_concentration: j1
+        irf: irf1
+    dataset4: &dataset4
+        megacomplex: [mc2]
+megacomplex:
+    mc1:
+        type: decay
+        k_matrix: [k1]
+    mc2:
+        type: spectral
+        shape:
+            s1: sh1
+            s2: sh2
+            s3: sh3
+irf:
```

Comment:
Can you add a case without IRF? Currently the full_model doesn't work without (see [fit_simData01.py](https://github.com/glotaran/pyglotaran-validation/blob/main/paramGUI/simData01/fit_simData01.py))

Reply:
So the problem is that the full models seem to be more sensible in paramters. There was a bug which led to that the result data was calculated with the parameters of last function eval, not the optimized. Ifixed this and it should work now. See also the unittest I added.